lambdas/service/app.py

```diff
@@ -546,204 +546,244 @@ def repository_search(entity_type: str, item_id: str):
         raise NotFoundError(msg=e)
 
 
-@app.route('/repository/files', methods=['HEAD', 'GET'], cors=True)
-@app.route('/repository/files/{file_id}', methods=['GET'], cors=True)
+generic_dict = schema.object(additional_properties=True)
+array_of_dict = schema.array(generic_dict)
+hit_schema = schema.object(
+    additional_properties=True,
+    protocols=array_of_dict,
+    entryId=str,
+    samples=array_of_dict,
+    specimens=array_of_dict,
+    cellLines=array_of_dict,
+    donorOrganisms=array_of_dict,
+    organoids=schema.array(str),
+    cellSuspensions=array_of_dict,
+)
+
+page_schema = schema.object(
+    hits=schema.array(hit_schema),
+    pagination=generic_dict,
+    termFacets=generic_dict
+)
+
+
+def repository_search_spec(entity_type):
+    return {
+        'tags': ['Repository'],
+        'parameters': [
+            params.query('filters',
+                         schema.optional(str),
+                         description='Filters to be applied when calling ElasticSearch.'),
+            params.query('size', schema.optional(int), description='Number of hits displayed per page.'),
+            params.query('sort', schema.optional(str), description='Which field to sort by.'),
```

Comment:
Full sentences, please, for description. What set of values is this recruited from? Can you make that more specific? The same applies to `filters` by the way.